pageserver/src/timelines.rs

```diff
@@ -37,6 +37,20 @@ pub fn get_info_from_loaded_timeline(
     include_non_incremental_logical_size: bool,
 ) -> anyhow::Result<LocalTimelineInfo> {
     let last_record_lsn = timeline.get_last_record_lsn();
+
+    let (wal_source_connstr, last_received_msg_lsn, last_received_msg_ts) = {
+        let guard = timeline.last_received_wal.lock().unwrap();
+        if let Some(info) = guard.as_ref() {
```

Comment:
Looks like we could leave all fields in our HTTP object without the `Option` wrapper and wrap the object itself in the timeline, as this `if let` logic implies.

Reply:
You mean, include the WalReceiverInfo struct itself as a field in LocalTimeLine, like this?
```
diff --git a/pageserver/src/http/models.rs b/pageserver/src/http/models.rs
index dea767ec..669ce724 100644
*** a/pageserver/src/http/models.rs
--- b/pageserver/src/http/models.rs
@@ -131,11 +131,17 @@
     pub current_physical_size_non_incremental: Option<u64>,
     pub timeline_state: LocalTimelineState,
 
-    pub wal_source_connstr: Option<String>,
+    pub wal_receiver_info: Option<WalReceiverInfo>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct WalReceiverInfo {
+    pub wal_source_connstr: String,
     #[serde_as(as = "Option<DisplayFromStr>")]
-    pub last_received_msg_lsn: Option<Lsn>,
+    pub last_received_msg_lsn: Lsn,
     /// the timestamp (in microseconds) of the last received message
-    pub last_received_msg_ts: Option<u128>,
+    pub last_received_msg_ts: u128,
 }
 
 #[serde_as]
```
Yeah, that would work. A flat response object somehow feels nicer to me, though. I can't really articulate why.
,

Reply:
Yes, but no objections against other variant, since there's no API user that needs this info now, anything works.